diff --git a/BUILD b/BUILD
new file mode 100644
index 00000000..ffffffff
--- /dev/null
+++ b/BUILD
@@ -0,0 +1,13 @@
+cc_library(
+    name = "aws-c-sdkutils",
+    srcs = glob(["source/*.c"]),
+    hdrs = glob([
+        "include/aws/sdkutils/*.h",
+        "include/aws/sdkutils/private/*.h",
+    ]),
+    includes = ["include"],
+    deps = [
+        "@aws-c-common",
+    ],
+    visibility = ["//visibility:public"],
+)
