diff --git a/BUILD b/BUILD
new file mode 100644
index 00000000..ffffffff
--- /dev/null
+++ b/BUILD
@@ -0,0 +1,8 @@
+cc_library(
+    name = "aws-sdk-s3",
+    srcs = glob(["generated/src/aws-cpp-sdk-s3/source/**/*.cpp"]),
+    hdrs = glob(["generated/src/aws-cpp-sdk-s3/include/**/*.h"]),
+    includes = ["generated/src/aws-cpp-sdk-s3/include"],
+    deps = ["@aws-sdk-core"],
+    visibility = ["//visibility:public"],
+)
