diff --git a/BUILD b/BUILD
new file mode 100644
index 00000000..ffffffff
--- /dev/null
+++ b/BUILD
@@ -0,0 +1,14 @@
+load("@rules_foreign_cc//foreign_cc:configure.bzl", "configure_make")
+
+filegroup(
+    name = "bin_source",
+    srcs = glob(["**"]),
+)
+
+configure_make(
+    name = "bison",
+    copts = ["-Wno-attributes"],
+    lib_source = ":bin_source",
+    out_binaries = ["bison"],
+    visibility = ["//visibility:public"],
+)
